src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -194,6 +219,53 @@ def _migrate_external_table(self, src_table: Table, rule: Rule, grants: list[Gra
         self._backend.execute(src_table.sql_alter_from(rule.as_uc_table_key, self._ws.get_workspace_id()))
         return self._migrate_acl(src_table, rule, grants)
 
+    def _migrate_external_table_hiveserde(
+        self,
+        src_table: Table,
+        rule: Rule,
+        grants: list[Grant] | None = None,
+        mounts: Iterable[Mount] | None = None,
+        hiveserde_in_place_migrate: HiveSerdeType | None = None,
+    ):
+        if not hiveserde_in_place_migrate:
+            # TODO: Add sql_migrate_external_hiveserde_ctas here
+            return False
+
+        # verify hive serde type
+        hiveserde_type = src_table.hiveserde_type(self._backend)
+        if hiveserde_type in [
+            HiveSerdeType.NOT_HIVESERDE,
+            HiveSerdeType.OTHER_HIVESERDE,
+            HiveSerdeType.INVALID_HIVESERDE_INFO,
+        ]:
+            logger.warning(f"{src_table.key} table can only be migrated using CTAS.")
+            return False
+        if hiveserde_type != hiveserde_in_place_migrate:
+            logger.info(
+                f"{src_table.key} is using {hiveserde_type.name} hiveserde. It won't be in-place migrated in this {hiveserde_in_place_migrate} dedicated task."
+            )
+            return False
+
+        # if the src table location is using mount, resolve the mount location so it will be used in the updated DDL
+        dst_table_location = None
+        if mounts and src_table.is_dbfs_mnt:
+            dst_table_location = ExternalLocations.resolve_mount(src_table.location, mounts)
```

Comment:
don't we need to eagerly resolve mounts for delta tables as well?

Reply:
for delta we use SYNC, and SYNC handles the mounts already